src/handlers/relink.rs

```diff
@@ -0,0 +1,101 @@
+//! This handler is used to "relink" linked GitHub issues into their long form
+//! so that when pulling subtree into the main repository we don't accidentaly
+//! closes issue in the wrong repository.
+//!
+//! Example: `Fixes #123` (in rust-lang/clippy) would now become `Fixes rust-lang/clippy#123`
+
+use std::borrow::Cow;
+
+use regex::Regex;
+
+use crate::{
+    config::RelinkConfig,
+    github::{Event, IssuesAction, IssuesEvent},
+    handlers::Context,
+};
+
+lazy_static! {
+    // Taken from https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue?quot#linking-a-pull-request-to-an-issue-using-a-keyword
+    static ref LINKED_RE: Regex =
+        Regex::new("(?i)(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved) +(#[0-9]+)").unwrap();
+}
```

Comment:
Nit:

since `lazy_static` [has been deprecated](https://github.com/rust-lang-nursery/lazy-static.rs/issues/214), when writing new code I personally prefer using the new [LazyLock](https://doc.rust-lang.org/std/sync/struct.LazyLock.html) API (where appropriate) so that we start slowing phasing out that crate.

wdy?

Reply:
Since we use `lazy_static` I didn't wanted to changed what we use, but since you asked I've changed all the places to use `LazyLock`.

Reply:
Well :sweat_smile:

I didn't exactly ask to replace the crate everywhere, that's would be a change out of scope in this patch. I was just thinking to use it *here* and in your patch (then maybe one day in another patch remove the `lazy_static` crate). I didn't mean to get you do unnecessary work - sorry if I wasn't clear :slightly_smiling_face:

Reply:
> I didn't exactly ask to replace the crate everywhere

I didn't meant to imply that, sorry.

I extracted the removal in https://github.com/rust-lang/triagebot/pull/1898.